conda/base/context.py

```diff
@@ -936,23 +948,6 @@ def channels(self):
             else:
                 return tuple(IndexedSet((*local_add, *self._argparse_args["channel"])))
 
-        # add 'defaults' channel when necessary if --channel is given via the command line
-        if self._argparse_args and "channel" in self._argparse_args:
-            # TODO: it's args.channel right now, not channels
-            argparse_channels = tuple(self._argparse_args["channel"] or ())
-            # Add condition to make sure that sure that we add the 'defaults'
-            # channel only when no channels are defined in condarc
-            # We needs to get the config_files and then check that they
-            # don't define channels
-            channel_in_config_files = any(
-                "channels" in context.raw_data[rc_file].keys()
-                for rc_file in self.config_files
-            )
-            if argparse_channels and not channel_in_config_files:
-                return tuple(
-                    IndexedSet((*local_add, *argparse_channels, DEFAULTS_CHANNEL_NAME))
-                )
```

Comment:
I'm thinking removing this outright might surprise users, and we probably should raise a [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning) about it. @kenodegard would you say it would suffice to add a similar `Channel list is empty. Please run 'conda config --add channels %s' to ...` warning here for one or two releases?

Reply:
I'll proceed the warning assuming Ken is alright with this direction.

Reply:
@jaimergp you should use `deprecated.topic` for this

Reply:
I've replaced them accordingly but now I don't get the warnings in the CLI 🤔 

This is what I'm using:

```python
if self._channels:
    _channels = self._channels
else:
    deprecated.topic(
        "24.9",
        "25.1",
        topic=f"Adding '{DEFAULTS_CHANNEL_NAME}' to channel list implicitly.",
        addendum="To remove this warning, please choose a default channel explicitly "
        "via 'conda config --add channels <name>'. In the future, the implicit "
        "default channel configuration will be removed.",
    )
    _channels = [DEFAULTS_CHANNEL_NAME]
```

Reply:
by default `deprecated.topic` uses a DeprecationWarning which is ignored by default (since its a developer warning, not a user warning)

you also need to specify `deprecation_type=FutureWarning`:

```python
if self._channels:
    _channels = self._channels
else:
    deprecated.topic(
        "24.9",
        "25.1",
        topic=f"Adding '{DEFAULTS_CHANNEL_NAME}' to channel list implicitly.",
        addendum=(
            "To remove this warning, please choose a default channel explicitly "
            "via 'conda config --add channels <name>'. In the future, the implicit "
            "default channel configuration will be removed."
        ),
        deprecation_type=FutureWarning,
    )
    _channels = [DEFAULTS_CHANNEL_NAME]
```

Reply:
I still don't see it but I think it's due to this clause in the generate_message logic, right?

```
        if self._version_less_than(deprecate_in):
            category = PendingDeprecationWarning
            warning = f"is pending deprecation and will be removed in {remove_in}."
```

Reply:
correct, while we are in the pending deprecation window you will need to explicitly enable all warnings to see the warnings (e.g., `PYTHONWARNINGS=d`), since `conda 24.9` hasn't been tagged yet the deprecation logic still interprets this as being a pending deprecation window